src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -125,6 +156,27 @@ private static CSharpCompilation MetadataView(CSharpCompilation comp, params Met
             return CreateCompilation("", references: references);
         }
 
+        private static CSharpCompilation RetargetedView(CSharpCompilation comp, params MetadataReference[] references)
+        {
+            var someType = comp.GlobalNamespace.GetMembers().OfType<NamedTypeSymbol>().FirstOrDefault();
+            references = references.Append(comp.ToMetadataReference());
+            var result = CreateCompilation("", references: references, targetFramework: TargetFramework.NetStandardLatest);
+            result.VerifyDiagnostics();
+            var originalCorLib = comp.Assembly.CorLibrary;
+            var newCorLib = result.Assembly.CorLibrary;
+            Assert.NotEqual(originalCorLib, newCorLib);
+
+            if (someType is { })
+            {
+                // confirm that it is retargeted
```

Comment:
>confirm that it is retargeted [](start = 19, length = 29)

Only types declared in ```comp.SourceModule``` are going to be retargeting. Consider adjusting the way we locate the type to make more precise and to ensure we are not going to pick up a type from a referenced assembly.